### ISSUE TITLE
Can't Stub attribute values with Aws::DynamoDB::Client.

### ISSUE THREAD

**Issue by trevorrowe** [aws/aws-sdk-ruby#770]
See original issue for more information: https://github.com/aws/aws-sdk-core-ruby/issues/218

**Comment by trevorrowe** [comment]
I spent some time working on this. I started by working on a patch that would convert the simplified attribute value formats back into the serialized format with types. This would ensure the response stubbing plugin would succeed at validating the data. Followed by that I had to then also reverse the data format because the normal response handlers were not firing. This was a code-smell to me.

The response stubbing plugin correctly exercises all of the request handlers that build the request. Once it reaches the send handler it simply sets the response data and then returns. This prevents any of response handlers from  being exercised. 

There was a related issue in the previous repo (https://github.com/aws/aws-sdk-core-ruby/issues/197) where the response stub skipped over the response handlers, preventing stubs that sets a response target from sending data to the target IO object.  This specific limitation was addressed then. I can certainly address this new bug with not supporting attribute values but then I feel like there will be something else in the future.

I think the correct way to resolve this issue is to re-work the response stubbing to take the stub data and to marshal it down to an actual HTTP response with headers, status code and a body. By doing this all of the response handlers will fire as normally ensuring that all of the service customizations work as intended.

I'm jotting down these notes here. I'm not sure I have time to do the feature work required right now and I want a log of what I discovered.
